moo.js

```diff
@@ -138,14 +144,44 @@
     var flags = hasSticky ? 'ym' : 'gm'
     var regexp = new RegExp(reUnion(parts) + suffix, flags)
 
-    return new Lexer(regexp, groups)
+    return {regexp: regexp, groups: groups}
   }
 
+  function compile(rules) {
+    var result = compileRules(rules)
+    return new Lexer({start: result}, 'start')
+  }
+
+  function compileStates(states, start) {
+    var keys = Object.getOwnPropertyNames(states)
+    if (!start) start = keys[0]
+
+    var map = {}
```

Comment:
It doesn't make a difference behavior-wise, but I think it is a bit clearer to use `Object.create(null)`